src/MessagePack.AspNetCoreMvcFormatter/MessagePackOutputFormatter.cs

```diff
@@ -20,23 +19,12 @@ public MessagePackOutputFormatter()
         public MessagePackOutputFormatter(MessagePackSerializerOptions options)
         {
             this.options = options;
-        }
-
-        public bool CanWriteResult(OutputFormatterCanWriteContext context)
-        {
-            if (!context.ContentType.HasValue)
-            {
-                context.ContentType = new StringSegment(ContentType);
-                return true;
-            }
 
-            return context.ContentType.Value == ContentType;
+            SupportedMediaTypes.Add(ContentType);
         }
 
-        public Task WriteAsync(OutputFormatterWriteContext context)
+        public override Task WriteResponseBodyAsync(OutputFormatterWriteContext context)
         {
-            context.HttpContext.Response.ContentType = ContentType;
```

Comment:
This logic is [part of the OutputFormatter](https://source.dot.net/#Microsoft.AspNetCore.Mvc.Core/Formatters/OutputFormatter.cs,186)